.github/workflows/security.yml

```
@@ -276,3 +284,291 @@
+      - name: Upload Scorecard results to GitHub Security
+        uses: github/codeql-action/upload-sarif@4e94bd11f71e507f7f87df81788dff88d1dacbfb  # v4.31.0
+        if: always()  # Upload even if score is below threshold
+        with:
+          sarif_file: scorecard-results.sarif
+          category: 'scorecard'
+
+      # Parse Scorecard results and validate against target threshold
+      # Fails workflow if score is below 9.0/10
+      - name: Validate Scorecard results
+        if: always()
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2086:info:1:40: Double quote to prevent globbing and word splitting

(shellcheck)